### PR TITLE
feat(unsubscribe): merge n1-unsubscribe plugin

### PR DESCRIPTION
**Pull request by KyleKing**
This PR will merge the n1-unsubscribe plugin into an internal plugin within nylas-mail

The plugin both unsubscribes and archives emails with a single button press. Users have the option of setting the default action to trash emails instead of archiving, to set a keyboard shortcut, add an optional confirmation dialog, and there is decent multilingual support.

The PR to merge the unsubscribe plugin is well overdue and will close https://github.com/nylas/nylas-mail/pull/3154, which happened right at the cusp of the N1 > Nylas Mail transition and the PR that occurred right at the official Nylas Team > OS Community transition: https://github.com/nylas/nylas-mail/pull/3612

### CODE REVIEW

File: packages/client-app/src/flux/modules/reflux-coffee.coffee [hunk 1]
```diff
@@ -153,7 +153,7 @@ module.exports =
     setupEmitter: ->
       return if @_emitter
       @_emitter ?= new EventEmitter()
-      @_emitter.setMaxListeners(50)
+      @_emitter.setMaxListeners(100)
```
Comment (dweremeichik) [inline packages/client-app/src/flux/modules/reflux-coffee.coffee]: Just curios why the listener limit needed to be raised?

**Comment by seesemichaelj** [comment]
@KyleKing could you make an issue on this repo for this? I know you already have two (that you have most graciously linked), but due to preservation sake I'd prefer the redundancy. Thanks!

On another note, as a built in plugin, this should not change any behavior for deleting or archiving emails correct? This should be supplemental buttons/keymaps that allow this enhanced feature.

**Comment by dweremeichik** [comment]
@seesemichaelj  the code appears to be scoped to the "new" internal _unsubscribe_ package.

File: packages/client-app/src/flux/modules/reflux-coffee.coffee (hunk 1 above)
Comment (KyleKing) [inline packages/client-app/src/flux/modules/reflux-coffee.coffee]: @dweremeichik I had to raise the maximum to allow more for the unsubscribe plugin. There doesn't appear to be any memory leaks and is just the nature of how the plugin was implemented

**Comment by KyleKing** [comment]
Wow, Nylas Mail is still *alive*! Thanks for the quick responses. The plugin will call the Nylas API for archiving or deleting, so there is no impact on any of the default Nylas behavior

@colinking and I are both welcome to input, so let us know what can be improved

**Pull request merged**